grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -136,26 +136,26 @@ static void attachGeneratedRelations(Representation currentHal, Map.Entry<VarNam
     private static void attachSingleGeneratedRelation(Representation currentHal, Concept currentVar, Concept otherVar, Map<VarName, String> roleTypes, VarName currentVarName, VarName otherVarName, Optional<TypeName> relationType, String keyspace) {
         ConceptId currentID = currentVar.getId();
 
-        ConceptId firstID;
-        ConceptId secondID;
+        String firstID;
```

Comment:
What was the reason for changing to the `String` ? 

This change has result in `getValue` being called more often than it needs to be.